wolfssl/options.h.in

```diff
@@ -30,6 +30,342 @@
 extern "C" {
 #endif
 
+#ifndef WOLFSSL_OPTIONS_IGNORE_SYS
```

Comment:
How did you generate this list of options? I need to understand the future maintenance of this list.

Reply:
I searched `CMakeLists.txt` for `list(APPEND WOLFSSL_DEFINITIONS`. In the future each option will need to be added manually to `options.h.in`.

Reply:
@SparkiDev or @JacobBarthelmeh thoughts about this improvement to CMake and options.h? It does require the options are added to options.h.in.

Reply:
This looks like what I saw in libsrtp.
Change makes sense to me.

Reply:
@JacobBarthelmeh once this is merged please update the release process to include checking for new / changed options.

Reply:
Thanks Sean and Jacob.

@oltolm did you add the options manually or use a script? Are you able to share a script to help extract the define list?

If cross compiling users might need to define `WOLFSSL_OPTIONS_IGNORE_SYS`, right?

Thanks,
David Garske, wolfSSL

Reply:
I added the options manually.

Reply:
Can you please move this cmake specific options template into a new file in cmake/options.h.in and not modify this .in? This isn't currently an issue modifying this file, but it could be.